bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/ChromecastCommander.java

```diff
@@ -229,44 +222,69 @@ private void handleMute(final Command command) {
         }
     }
 
-    public void playMedia(@Nullable String title, @Nullable String url, @Nullable String mimeType) {
+    public void startApp(@Nullable String appId) {
+        if (appId == null) {
+            return;
+        }
         try {
-            if (chromeCast.isAppAvailable(MEDIA_PLAYER)) {
-                if (!chromeCast.isAppRunning(MEDIA_PLAYER)) {
-                    final Application app = chromeCast.launchApp(MEDIA_PLAYER);
+            if (chromeCast.isAppAvailable(appId)) {
+                if (!chromeCast.isAppRunning(appId)) {
+                    final Application app = chromeCast.launchApp(appId);
                     statusUpdater.setAppSessionId(app.sessionId);
-                    logger.debug("Application launched: {}", app);
-                }
-                if (url != null) {
-                    // If the current track is paused, launching a new request results in nothing happening, therefore
-                    // resume current track.
-                    MediaStatus ms = chromeCast.getMediaStatus();
-                    if (ms != null && MediaStatus.PlayerState.PAUSED == ms.playerState && url.equals(ms.media.url)) {
-                        logger.debug("Current stream paused, resuming");
-                        chromeCast.play();
-                    } else {
-                        chromeCast.load(title, null, url, mimeType);
-                    }
+                    logger.debug("Application launched: {}", appId);
                 }
             } else {
-                logger.warn("Missing media player app - cannot process media.");
+                logger.warn("Failed starting app, app probably not installed. Appid: {}", appId);
             }
             statusUpdater.updateStatus(ThingStatus.ONLINE);
```

Comment:
Should it be kept there or only in playMedia ?

Reply:
I extracted the starting of the app out of the playMedia method. This way it would be possible to use the same routine to start different apps. So i prefer to leave it here.